src/modules/ekf2/EKF/ekf_helper.cpp

```diff
@@ -483,18 +483,29 @@ bool Ekf::realignYawGPS()
 }
 
 // Reset heading and magnetic field states
-bool Ekf::resetMagHeading(const Vector3f &mag_init, bool increase_yaw_var, bool update_buffer)
+bool Ekf::resetMagHeading(bool increase_yaw_var, bool update_buffer)
 {
 	// prevent a reset being performed more than once on the same frame
 	if (_imu_sample_delayed.time_us == _flt_mag_align_start_time) {
 		return true;
 	}
 
+	const Vector3f mag_init = _mag_lpf.getState();
+
+	constexpr float minimum_earth_mag_field_strength = 0.25f; // Gauss
+
+	if (mag_init.norm() < minimum_earth_mag_field_strength / 2.f) {
```

Comment:
Better way to do this?

Reply:
can't we use `_control_status.flags.mag_field_disturbed` ?

Reply:
I meant as a simple check that _mag_lpf is available. I updated it to use `_mag_counter`.